### PR TITLE
dealii: fix concretization problem between trilinos and adol-c

### LINKED ISSUE #20445: Problems during install deal.II in spack
Hi everyone, first I want to apologize because I am new into ubuntu, so I could be wrong anywhere.

In order to install a library, I had to clone spack, and then to install deal.II and vtk. 
I followed this code: 

```
.bashrc file: "export SPACK_ROOT=/path/to/spack/
export PATH=${SPACK_ROOT}/bin:${PATH}"
sudo apt install python git curl make g++ gfortran
mkdir -p ${SPACK_ROOT}
cd ${SPACK_ROOT}
git clone https://github.com/spack/spack.git .
spack install environment-modules
.bashrc file: "source $(spack location -i environment-modules)/init/bash
source ${SPACK_ROOT}/share/spack/setup-env.sh"
```

Up to this point, everything seems fine, and I got no errors. 
Then I install 

`spack install dealii@9.2.0 ^suite-sparse@5.2.0`

And I get errors as you see in the screen. 
Do you have any ideas about what could I do? Thanks everyone! 
errore dealii

### PR DESCRIPTION
**Pull request by luca-heltai**
Fixes #20445

### CODE REVIEW

File: var/spack/repos/builtin/packages/dealii/package.py [hunk 1]
```diff
@@ -159,6 +159,8 @@ class Dealii(CMakePackage, CudaPackage):
     depends_on('ginkgo',           when='@9.1:+ginkgo')
     depends_on('gmsh+tetgen+netgen+oce', when='@9.0:+gmsh', type=('build', 'run'))
     depends_on('gsl',              when='@8.5.0:+gsl')
+    # FIXME: next line fixes concretization with trilinos and adol-c
+    depends_on('trilinos~exodus~netcdf',    when='@9.0:+adol-c')
```
Comment (jppelteret) [inline var/spack/repos/builtin/packages/dealii/package.py]: Makes sense, due to the added the conflicts. But shouldn't this be `when='@9.0:+adol-c+trilinos'`? 

Also, could we please put this further down the list (~line 199 with the other trilinos depends)? That way this list remains (nearly) alphabetically ordered.

Comment (jppelteret) [reply var/spack/repos/builtin/packages/dealii/package.py]: Thanks for the fix, BTW :-)

Comment (luca-heltai) [reply var/spack/repos/builtin/packages/dealii/package.py]: Thanks @jppelteret!

**Review by jppelteret — APPROVED** [review]
Thanks @luca-heltai! LGTM :-)

**Review by alalazo — APPROVED** [review]

**Pull request merged**